### PR TITLE
implement logging

### PR DESCRIPTION
**Pull request by filedesless**
depends on roadrunner-craft/core#23

#40

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ notify = { version = "5.0.0-pre.2", optional = true }
 math = { path = "../math" }
 #core = { git = "https://github.com/roadrunner-craft/core" }
 core = { path = "../core" }
+log = { version = "0.4.8", features = ["std"] }
```
Comment (afrigon) [inline Cargo.toml:22]: use re-export from the core instead of depending on the log crate here

Comment (filedesless) [reply Cargo.toml]: good call, did that

**Comment by Corb3nik** [comment]
LGTM

File: src/main.rs [hunk 2]
```diff
@@ -27,8 +31,15 @@ use std::time::{Duration, Instant};
...
 const FRAME_RATE_CAP: u32 = 60;
 const PKG_NAME: &'static str = env!("CARGO_PKG_NAME");
+const PKG_VERSION: &'static str = env!("CARGO_PKG_VERSION");
 
 fn main() -> io::Result<()> {
+    logging::init(vec![
+        Box::new(Logger::to_file("logs", Level::Debug)),
+        Box::new(Logger::to_stdout(Level::Info)),
+    ]);
+    info!("Welcome to {} v{}", PKG_NAME, PKG_VERSION);
```
Comment (afrigon) [inline src/main.rs:41]: met juste ça, fuck le welcome
`info!("{} v{}", PKG_NAME, PKG_VERSION);`

File: src/render/texture/texture.rs [hunk 3]
```diff
@@ -31,7 +32,7 @@ impl Texture {
 
                     let width = img.width();
                     if width != img.height() {
-                        println!("<texture> Image aspect ratio must be 1: {}", path);
+                        error!("<texture> Image aspect ratio must be 1: {}", path);
```
Comment (afrigon) [inline src/render/texture/texture.rs:35]: this should be a warn

File: src/render/texture/texture.rs [hunk 4]
```diff
@@ -20,7 +21,7 @@ impl Texture {
             None => Texture::default(),
             Some(path) => match image::open(path) {
                 Err(err) => {
-                    println!("<texture> Could not load image {}: {}", path, err);
+                    error!("<texture> Could not load image {}: {}", path, err);
```
Comment (afrigon) [inline src/render/texture/texture.rs:24]: this should also be a warn

File: src/render/texture/texture_array.rs [hunk 5]
```diff
@@ -64,7 +65,7 @@ impl TextureArray {
         if let Some(path) = path.to_str() {
             match image::open(path) {
                 Err(err) => {
-                    println!("<texture-array> Could not load image {}: {}", path, err);
+                    error!("<texture-array> Could not load image {}: {}", path, err);
                     // TODO: add default image
```
Comment (afrigon) [inline src/render/texture/texture_array.rs:69]: warn aussi pour c'est deux là

File: src/render/texture/texture_array.rs [hunk 6]
```diff
@@ -74,7 +75,7 @@ impl TextureArray {
                     };
 
                     if img.width() != self.size || img.height() != self.size {
-                        println!("<texture-array> Image aspect ratio must be 1: {}", path);
+                        error!("<texture-array> Image aspect ratio must be 1: {}", path);
```
Comment (afrigon) [inline src/render/texture/texture_array.rs:78]: warn here too

File: src/main.rs [hunk 7]
```diff
@@ -27,8 +31,15 @@ use std::time::{Duration, Instant};
 const FPS_REFRESH_TIMEOUT: u64 = 1;
 const FRAME_RATE_CAP: u32 = 60;
 const PKG_NAME: &'static str = env!("CARGO_PKG_NAME");
+const PKG_VERSION: &'static str = env!("CARGO_PKG_VERSION");
 
 fn main() -> io::Result<()> {
+    logging::init(vec![
+        Box::new(LogFile::new(Level::Debug)),
+        Box::new(LogStdOut::new(Level::Info)),
```
Comment (afrigon) [inline src/main.rs:39]: what should we do with the log level ? always the same with a verbose flag or should we set it according to debug/release build

**Comment by afrigon** [comment]
oops lmao I fucked up the commit history my bad. moved to #78

**Pull request closed without merging**